librz/core/cdebug.c

```diff
@@ -1165,3 +1165,96 @@ RZ_IPI bool rz_core_debug_desc_print(RzDebug *dbg, RzCmdStateOutput *state) {
 	rz_list_free(list);
 	return true;
 }
+
+struct RzCoreDebugState {
+	RzDebug *dbg;
+	RzCmdStateOutput *state;
+};
+
+static const char *signal_option(int opt) {
+	if (opt & RZ_DBG_SIGNAL_CONT) {
+		return ("cont");
+	}
+	if (opt & RZ_DBG_SIGNAL_SKIP) {
+		return ("skip");
+	}
+	return NULL;
+}
+
+static bool siglistcb(void *p, const char *k, const char *v) {
+	static char key[32] = "cfg.";
+	struct RzCoreDebugState *ds = p;
+	int opt;
+	if (atoi(k) > 0) {
+		strncpy(key + 4, k, 20);
+		opt = sdb_num_get(ds->dbg->sgnls, key, 0);
+		if (opt && ds->state->mode == RZ_OUTPUT_MODE_STANDARD) {
+			rz_cons_printf("%s %s", k, v);
+			const char *optstr = signal_option(opt);
+			if (optstr) {
+				rz_cons_printf(" %s", optstr);
+			}
+			rz_cons_newline();
+		} else {
+			rz_cons_printf("%s %s\n", k, v);
+		}
+	}
+	return true;
+}
+
+static bool siglistjsoncb(void *p, const char *k, const char *v) {
+	static char key[32] = "cfg.";
+	struct RzCoreDebugState *ds = p;
+	int opt;
+	if (atoi(k) > 0) {
+		strncpy(key + 4, k, 20);
+		opt = (int)sdb_num_get(ds->dbg->sgnls, key, 0);
+		pj_o(ds->state->d.pj);
+		pj_ks(ds->state->d.pj, "signum", k);
```

Comment:
should this be a `_kn` instead of a string?

Reply:
`k` in this code is `char *`, thus it's `pj_ks`.

Reply:
i know, but we should probably provide the number representation instead.

Reply:
how about we do both?